### ISSUE TITLE
Fix false positives for scientific notation (ex: 9e9) in unit-no-unknown

### ISSUE THREAD

**Issue by chearon** [stylelint/stylelint#3537]
> Clearly describe the bug

`no-unit-unknown` says `Unexpected unknown unit "e9"` when parsing a rule like `flex-shrink: 9e9`

It should eat up the `e9` before the unit like the spec says for [`number-ident`](https://www.w3.org/TR/css-syntax-3/#typedef-number-token)

> Which rule, if any, is the bug related to?

`no-unit-unknown`

> What CSS is needed to reproduce the bug?

```css
a {
  flex: 0 9e9 auto;
  width: 1e5px;
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "no-unit-unknown": "true"
  }
}
```

> Which version of stylelint are you using?

`9.1.2`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

`yarn stylelint style.css`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

```shell
 460:10  ✖  Unexpected unknown unit "e9"   unit-no-unknown
```

**Comment by jeddy3** [comment]
@chearon Thanks for the report and for using the template.

I've not seen that notation used in CSS before. It's fascinating to see! Anyhow, I've [raised an issue](https://github.com/TrySound/postcss-value-parser/issues/37) upstream in the value parser that we use in stylelint. Let's wait to see whether this bug will be addressed there, or whether we'll need to accommodate it directly in stylelint.

**Comment by jeddy3** [comment]
@chearon The upstream parser [isn't actively being maintained, but will accept a PR](https://github.com/TrySound/postcss-value-parser/issues/37#issuecomment-413131431). Alternatively, you can adapt the rule to use [postcss-values-parser](https://www.npmjs.com/package/postcss-values-parser), an alternative value parser. This parser does support scientific notation. You can walk through the `number` nodes and check the value of the `unit` property. [Here's an example](https://github.com/barnardos/stylelint-config-barnardos/blob/d78e96e9418dc02bd6a80a8711708a2155af1e1e/plugins/stylelint-color-hex-whitelist/index.js#L23-L29) of walking `word` nodes. It's a good first issue and I'm going to label accordingly.

As the issue has been labelled "help wanted", please consider [contributing a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule).

**Comment by chearon** [comment]
Hmm I'd hate to make you have both both `postcss-value-parser` and `postcss-values-parser` in your deps though! It's easy to fix in the former, so I'll submit a PR there shortly.

**Comment by jeddy3** [comment]
>  so I'll submit a PR there shortly.

Nice!

**Comment by evilebottnawi** [comment]
Fixed in https://github.com/TrySound/postcss-value-parser/releases/tag/v3.3.1, need add tests and update deps
